Added object and endpoints for managed credentials

yoshizzle opened the pull request:
GET all credentials and specific credentials by ID
POST to create a new credential
PUT to update an existing credential
POST to password to change an existing credential's password
POST to revoke to remove a credential

Commit:
Added object and endpoints for managed credentials

GET all credentials and specific credentials by ID
POST to create a new credential
PUT to update an existing credential
POST to password to change an existing credential's password
POST to revoke to remove a credential

SzySteve commented on docs/src/data/endpoints/managed.yaml:
```diff
@@ -175,3 +175,104 @@ endpoints:
+              A unique label describing the credential being added.
+          username:
+            type: String
+            description: >
```
optional: True

SzySteve commented on docs/src/data/endpoints/managed.yaml:
```diff
@@ -175,3 +175,104 @@ endpoints:
+              An optional login username.
+          password:
+            type: String
+            description: >
```
optional: True

yoshizzle replied:
password is required.

SzySteve commented on docs/src/data/endpoints/managed.yaml:
```diff
@@ -175,3 +175,104 @@ endpoints:
+            curl -H "Authorization: Bearer $TOKEN" \
+                https://$api_root/$version/managed/credential/$credential_id
+      PUT:
+        description: >
```
Should note what params are optional?

Dorthu replied:
I don't think so, PUTs are at least partially generated from the object I think

yoshizzle replied:
Not sure what you mean. I don't think we do that typically.

Dorthu commented on docs/src/data/objects/managedcredential.yaml:
```diff
@@ -0,0 +1,35 @@
+  last_decryption:
+    type: Object
+    description: >
+      The username and datetime of the last successful login using this credential.
```
not sure if these are "successful" decryptions only

yoshizzle replied:
I will remove the word successful.

Commit:
Addressed feedback PR

password is required
removed "successful" from login description in object
added optional true for username
